### PR TITLE
git: add a `gc` command

### PR DESCRIPTION
**Pull request by strib**
r? @oconnor663 

In the near future, git commands on keybase remotes will sometimes recommend users run `keybase git gc` to optimize their repos, and clean out old files.  Right now all it will do it pack refs, but later we will add more options for pruning old objects and re-packing objects.

Though the service<->KBFS protocol will support specifying thresholds for garbage collection (like the real git commands do via the git config file), for now this command hides it all from the user behind a single "--force" flag, and if that's not present it uses hard-coded defaults for any thresholds.

If the commands end up taking too long, we might eventually want to pipe status output back from the KBFS process, to show users on the command line during execution, but for now it just says "This may take several minutes...".

Issue: KBFS-2540

### CODE REVIEW

**Comment by strib** [comment]
circle will fail until I implement the new RPC on the KBFS side.  I'll do that before I merge this and merge them together.

File: go/client/cmd_git_gc.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	teamName keybase1.TeamName
+	force    bool
+}
+
+func newCmdGitGC(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	return cli.Command{
+		Name:         "gc",
+		ArgumentHelp: "<repo name> [--team=<team name>]",
+		Usage:        "Run garbage collection on a personal or team git repository.",
+		Description:  "`keybase git gc reponame` will run garbage collection on  a personal git\n   repo. `keybase git create reponame --team=treehouse` will run garbage\n   collection on a team git repo for the `treehouse` team.  Garbage collection\n   cleans up unnecessary files and optimizes the local repo.",
```
Comment (oconnor663) [inline go/client/cmd_git_gc.go]: What does "optimizes the local repo" mean? Does this mean that my local clone of the Keybase repo gets cleaned up too? (It doesn't sound like this command needs to be run inside my local clone though.)

Comment (strib) [reply go/client/cmd_git_gc.go]: Oops, that's a typo, it's not supposed to say "local".  I copied that from the `git-gc` man page.

File: go/service/git.go [hunk 2]
```diff
@@ -243,6 +247,79 @@ func (h *GitHandler) DeleteTeamRepo(ctx context.Context, arg keybase1.DeleteTeam
...
+		return err
+	}
+	folder := keybase1.Folder{
+		Name:       h.G().Env.GetUsername().String(),
+		FolderType: keybase1.FolderType_PRIVATE,
+		Private:    true,
+	}
+	options := keybase1.GcOptions{}
+	if !arg.Force {
+		options.MaxLooseRefs = gitDefaultMaxLooseRefs
```
Comment (oconnor663) [inline go/service/git.go]: Just confirming, zero in this field means "definitely always gc"?

Comment (strib) [reply go/service/git.go]: Yep!

**Pull request merged**